.github/workflows/ubuntu.yml

```diff
@@ -58,7 +61,9 @@ jobs:
           CCACHE_DIR=$GITHUB_WORKSPACE/.ccache ccache -sz -M 5G
 
       - name: Build and install folly and fmt
-        run: CMAKE_C_COMPILER_LAUNCHER=ccache CMAKE_CXX_COMPILER_LAUNCHER=ccache CCACHE_DIR=$GITHUB_WORKSPACE/.ccache scripts/setup-ubuntu.sh
+        # sudo doesn't preserve environment vairable; set it after sudo: https://stackoverflow.com/questions/8633461/how-to-keep-environment-variables-when-using-sudo/33183620#33183620
+        run: |
+          sudo CMAKE_C_COMPILER_LAUNCHER=ccache CMAKE_CXX_COMPILER_LAUNCHER=ccache CCACHE_DIR=$GITHUB_WORKSPACE/.ccache scripts/setup-ubuntu.sh
```

Comment:
Can this work? https://stackoverflow.com/questions/8633461/how-to-keep-environment-variables-when-using-sudo

Reply:
@ejguan : This is the same question in the link , but I choose a different answer lol.